src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Alright, sounds good and no rush!

One last thing though, your example has `C` as the last child of `F` and mine has it as the first child of `F`.  On one hand I prefer mine because it's a very strict interpretation of `swap`, but on the other hand, your example doesn't disturb the places of `L` and `M` since it pushes `C` on the end of the array.

Not sure which way to go with this part because those are both desirable behaviors.